src/context/WidgetSettingContext.tsx

```diff
@@ -85,10 +85,18 @@ export const WidgetSettingProvider = ({
       botId,
     });
 
-    const reuseCachedSession =
-      !!cachedSession &&
-      cachedSession.strategy === strategy &&
-      !isPastTime(cachedSession.expireAt);
+    const reuseCachedSession = ((cache: typeof cachedSession): cache is NonNullable<typeof cachedSession> => {
+      if (!cache || cache.strategy !== strategy) return false;
+      if (cache.strategy === 'manual') {
+        // NOTE: There is no need to check the expiration of the session if it is managed manually.
+        // However, since the existing logic has been regenerating the channel every 30 days due to this logic.
+        return !isPastTime(cache.expireAt) && cache.userId === injectedUserId;
+      }
+      if (cache.strategy === 'auto') {
+        return !isPastTime(cache.expireAt);
+      }
```

Comment:
```suggestion
      const isValidExpiration = !isPastTime(cache.expireAt);
      const isUserMatching = cache.userId === injectedUserId;
      if (cache.strategy === 'auto') return isValidExpiration;
      // NOTE: There is no need to check the expiration of the session if it is managed manually.
      // However, since the existing logic has been regenerating the channel every 30 days due to this logic.
      if (cache.strategy === 'manual') return isValidExpiration && isUserMatching;
```

nit; suggesting this since it seems you prefer a single line if statement ;)